src/Executables/ExportCoordinates/ExportCoordinates.hpp

```diff
@@ -110,8 +114,148 @@ struct ExportCoordinates {
     return std::forward_as_tuple(std::move(box));
   }
 };
+
+struct PrintMinimumGridSpacing {
+  template <typename ParallelComponent, typename DbTags, typename Metavariables,
+            typename ArrayIndex>
+  static void apply(db::DataBox<DbTags>& /*box*/,
+                    const Parallel::CBase_GlobalCache<Metavariables>& /*cache*/,
+                    const ArrayIndex& /*array_index*/,
+                    const double& overall_min_grid_spacing) noexcept {
+    printf("Overall inertial minimum grid spacing: %f\n\n",
+           overall_min_grid_spacing);
+  }
+};
 }  // namespace Actions
 
+template <size_t Dim, typename Metavars>
+struct SingletonParallelComponent {
+  using chare_type = Parallel::Algorithms::Singleton;
+  using const_global_cache_tag_list = tmpl::list<>;
+  using metavariables = Metavars;
+  using phase_dependent_action_list = tmpl::list<Parallel::PhaseActions<
+      typename Metavars::Phase, Metavars::Phase::Initialization, tmpl::list<>>>;
+  using initialization_tags = Parallel::get_initialization_tags<
+      Parallel::get_initialization_actions_list<phase_dependent_action_list>>;
+
+  static void execute_next_phase(
+      const typename Metavars::Phase /*next_phase*/,
+      const Parallel::CProxy_GlobalCache<Metavars>& /*global_cache*/) {}
+};
+
+template <typename Metavars, typename PhaseDepActionList,
+          class ImportInitialData = ImportNoInitialData>
+struct ElementArray;
+
+template <size_t Dim, typename PhaseDepActionList, class ImportInitialData>
+struct ArrayReduce {
+  template <typename ParallelComponent, typename DbTags, typename Metavars,
+            typename ArrayIndex>
+  static void apply(const db::DataBox<DbTags>& box,
+                    const Parallel::GlobalCache<Metavars>& cache,
+                    const ArrayIndex& array_index) noexcept {
+    if constexpr (db::tag_is_retrievable_v<
+                      domain::Tags::MinimumGridSpacing<Dim, Frame::Inertial>,
+                      db::DataBox<DbTags>>) {
```

Comment:
This `if` statement converts a compile-time error to a runtime error. If this action would remain a simple action, I would suggest you instead handle the issue like the `ExportCoordinates` action above does, i.e. by using `Requires` to instantiate the `apply` function template only if the condition is fulfilled. But once you convert the action to an iterable action, you won't need the `if` anymore at all, so just remove it.